django/db/models/sql/subqueries.py

```
@@ -133,8 +137,8 @@ def get_related_updates(self):
         for model, values in self.related_updates.items():
             query = UpdateQuery(model)
             query.values = values
-            if self.related_ids is not None:
-                query.add_filter("pk__in", self.related_ids)
+            ids = self.related_ids.get(model, [])
```

[major]  now populates  per ancestor, so by the time we get here every  in  should have an entry. Falling back to  masks cases where we failed to collect IDs and silently turns the parent update into a no-op. Please require the key (e.g. ) or assert that it exists so we fail loudly when invariants break.